### ISSUE TITLE
JAX-RS sub-resource locators with parameter don't always match

### ISSUE THREAD

**Issue by Emmeral** [quarkusio/quarkus#26028]
### Describe the bug

When using JAX-RS resource locators and sub-resources I don't always get the expected behavior. The paths don't match as expected. If you have a resource locator annotated with `@Path("{id}SomethingElse")` and a nested path in the sub-resource, quarkus will not match the path but rather output a `404`.

I think this is related to [this](https://github.com/quarkusio/quarkus/blob/d5de3c342ddb3b622ec7c63fd1e42fdc985c108f/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/URITemplate.java#L172) line in the `URITemplate` class. Or more specifically that the `URITemplate` is mistakenly constructed with `prefixMatch` set to `false`. It should be set to `true` as the path only refers to a resource locator and not an actual method. Sadly I could not inspect further from where the constructor is called.

### Expected behavior

_Referring to code posted below_

If I request `GET /b/TomOther/greet` I expect to get returned `Hello Tom` (same as when calling `GET /a/Tom/greet`)

### Actual behavior

I get `404 - Resource Not Found` status code.

### How to Reproduce?

[Repository](https://github.com/Emmeral/quarkus-subresource-reproducer) or see code below

```java
import javax.ws.rs.Path;
import javax.ws.rs.container.ResourceContext;
import javax.ws.rs.core.Context;

@Path("/")
public class RootResource {

    @Context
    ResourceContext resourceContext;

    @Path("a/{id}")
    public SubResource sub() {
        return resourceContext.getResource(SubResource.class);
    }

    @Path("b/{id}Other")
    public SubResource sub2() {
        return resourceContext.getResource(SubResource.class);
    }
}

```

```java
import io.quarkus.arc.Unremovable;
import javax.enterprise.context.RequestScoped;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;


@Unremovable
@RequestScoped
@Produces("text/plain")
public class SubResource {



    @Path("/greet")
    @GET
    public String greet(@PathParam("id") String id){
        return "Hello " + id;
    }


}

```

### Output of `java -version`

openjdk version "11.0.13" 2021-10-19

### Quarkus version or git rev

2.9.2.Final

**Comment by geoand** [comment]
Since you have already analyzed the code, would you like to provide a Pull Request with your proposed change?

**Comment by Emmeral** [comment]
Sadly I did not find the origin of the constructor call. I think it is related to some code which is generated at build time, so my IDE could not resolve the source.

**Comment by Emmeral** [comment]
I already did in the original Issue.
But [here](https://github.com/Emmeral/quarkus-subresource-reproducer) is the link again.

**Comment by Emmeral** [comment]
I further looked into the issue and I think I found the responsible constructor call. I fixed it in #26115

**Comment by Emmeral** [comment]
Added a new PR to the correct base branch. See #26224
